trino/integration_test.go

```diff
@@ -1325,3 +1460,240 @@ func TestIntegrationLargeQuery(t *testing.T) {
+		&goRow,
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestIntegrationSelectTpch1000SpoolingSpolledSegmentJsonZstdEncoded(t *testing.T) {
+	db := integrationOpen(t)
+	defer db.Close()
+	rows, err := db.Query("SELECT * FROM tpch.sf1.customer LIMIT 1000", sql.Named("X-Trino-Query-Data-Encoding", "json+zstd"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer rows.Close()
+	count := 0
+	for rows.Next() {
+		count++
+		var col tpchRow
+		err = rows.Scan(
+			&col.CustKey,
+			&col.Name,
+			&col.Address,
+			&col.NationKey,
+			&col.Phone,
+			&col.AcctBal,
+			&col.MktSegment,
+			&col.Comment,
+		)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+	if rows.Err() != nil {
+		t.Fatal(err)
+	}
+	if count != 1000 {
+		t.Fatal("not enough rows returned:", count)
+	}
+}
+
+func TestIntegrationSelectTpch1000SpoolingSpolledSegmentJsonEncoded(t *testing.T) {
+	db := integrationOpen(t)
+	defer db.Close()
+	rows, err := db.Query("SELECT * FROM tpch.sf1.customer LIMIT 1000", sql.Named("X-Trino-Query-Data-Encoding", "json"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer rows.Close()
+	count := 0
+	for rows.Next() {
+		count++
+		var col tpchRow
+		err = rows.Scan(
+			&col.CustKey,
+			&col.Name,
+			&col.Address,
+			&col.NationKey,
+			&col.Phone,
+			&col.AcctBal,
+			&col.MktSegment,
+			&col.Comment,
+		)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+	if rows.Err() != nil {
+		t.Fatal(err)
+	}
+	if count != 1000 {
+		t.Fatal("not enough rows returned:", count)
+	}
+}
+
+func TestIntegrationSelectTpch1000SpoolingSpolledSegmentJsonLz4Encoded(t *testing.T) {
+	db := integrationOpen(t)
+	defer db.Close()
+	rows, err := db.Query("SELECT * FROM tpch.sf1.customer LIMIT 1000", sql.Named("X-Trino-Query-Data-Encoding", "json+lz4"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer rows.Close()
+	count := 0
+	for rows.Next() {
+		count++
+		var col tpchRow
+		err = rows.Scan(
+			&col.CustKey,
+			&col.Name,
+			&col.Address,
+			&col.NationKey,
+			&col.Phone,
+			&col.AcctBal,
+			&col.MktSegment,
+			&col.Comment,
+		)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+	if rows.Err() != nil {
+		t.Fatal(err)
+	}
+	if count != 1000 {
+		t.Fatal("not enough rows returned:", count)
+	}
+}
+
+func TestIntegrationSelectTpch100SpoolingInlineSegmentJsonZstdEncoded(t *testing.T) {
+	db := integrationOpen(t)
+	defer db.Close()
+	rows, err := db.Query("SELECT * FROM tpch.sf1.customer LIMIT 100", sql.Named("X-Trino-Query-Data-Encoding", "json+zstd"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer rows.Close()
+	count := 0
+	for rows.Next() {
+		count++
+		var col tpchRow
+		err = rows.Scan(
+			&col.CustKey,
+			&col.Name,
+			&col.Address,
+			&col.NationKey,
+			&col.Phone,
+			&col.AcctBal,
+			&col.MktSegment,
+			&col.Comment,
+		)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+	if rows.Err() != nil {
+		t.Fatal(err)
+	}
+	if count != 100 {
+		t.Fatal("not enough rows returned:", count)
+	}
+}
+
+func TestIntegrationSelectTpch100SpoolingInlineSegmentJsonlz4Encoded(t *testing.T) {
```

Comment:
The names of these tests should help developers understand why they're needed, they don't have to list all options used. If you're testing all supported compression methods, I'd suggest doing this in a single, tabular test, with a name that'd say so.